File: codes/Heap/Python/HeapMain.py
import HeapModule as hm

heap = hm.Heap(lambda ch1, ch2: ord(ch2)-ord(ch1))

heap.HInsert('A')
heap.HInsert('B')
heap.HInsert('C')
print(heap.HDelete())

heap.HInsert('A')
heap.HInsert('B')
heap.HInsert('C')
print(heap.HDelete())

while heap.HIsEmpty() is not True:
    print(heap.HDelete())

File: codes/Heap/Java/Heap/HeapMain.java
package Heap;

public class HeapMain {
    public static void main(String[] args) {
        Heap<Character> heap = new Heap<Character>((ch1, ch2)->ch2-ch1);

        heap.HInsert('A');
        heap.HInsert('B');
        heap.HInsert('C');
        System.out.println(heap.HDelete());

        heap.HInsert('A');
        heap.HInsert('B');
        heap.HInsert('C');
        System.out.println(heap.HDelete());

        while(!heap.HIsEmpty())
            System.out.println(heap.HDelete());
    }
}
